### ISSUE TITLE
Access Transparent Gateway through hostname

### ISSUE THREAD

**Issue by wvangeem** [Azure/iotedge#3732]
Hi,

I've succesfully set up a Transparent Gateway with a child device.
I can use the example EdgeDownStreamDevice, but only with this connection string:
`HostName=myhubname.azure-devices.net;DeviceId=iotdemo07-child01;SharedAccessKey=<MyKey>;GatewayHostName=iotdemo07`
When I stop the iotedge service on my gateway, the messages are not sent anymore, so it really uses my gateway.
When I use following connection string:
`HostName=iotdemo07;DeviceId=iotdemo07-child01;SharedAccessKey=<MyKey>`

The error "Missing IoT hub name' is given with the EdgeDownStreamDevice code.
Any idea?

Extra information: my gateway is authenticated with a DPS and then assigned an IoTHub. That's way I would like to avoid having IoTHub information in my connection string.

**Comment by massand** [comment]
@wvangeem thanks for your question. All Azure IoT SDKs require downstream devices, connecting to IoT Edge as their gateway device, to be pre-provisioned in IoT Hub and linked as a child device. The only acceptable connection string format for all SDKs  requires the `HostName` parameter to be the IoT Hub hostname. There is no way to provision a child device via DPS and assign it to a specific IoT Edge device. 

Please review this document and let me know if you have further questions - https://docs.microsoft.com/en-us/azure/iot-edge/how-to-connect-downstream-device

**Comment by wvangeem** [comment]
@massand thanks for the reply. My goal is not to pre-provision the child devices. I already saw that they could not be provisioned through DPS.
So they are created as child devices to my provisioned gateway.
When you check this document: [https://docs.microsoft.com/en-us/azure/iot-edge/how-to-authenticate-downstream-device](url)
It says:
Instead of: HostName=myiothub.azure-devices.net;DeviceId=myDownstreamDevice;SharedAccessKey=xxxyyyzzz;GatewayHostName=myGatewayDevice
Thanks to the parent/child relationship, you can simplify the connection string by calling the gateway directly as the connection host. For example: HostName=myGatewayDevice;DeviceId=myDownstreamDevice;SharedAccessKey=xxxyyyzzz

I would like to avoid having to configure my IoTHub name in my child devices.
And I tried this, and it doesn't work. So the question is: is this an error in the documentation, or does this not work when the gateway is provisioned through DPS? When the gateway is connecting, it receives the IoThub it is connecting too, so the child devices could know it too.

**Comment by massand** [comment]
@wvangeem thanks for the pointers. Just to be sure EdgeHub configuration checks out, please provide the output for the troubleshooting instructions in this article - https://docs.microsoft.com/en-us/azure/iot-edge/how-to-connect-downstream-device#test-the-gateway-connection. 

It appears your error is emerging from the SDK during [connection string validation](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/IotHubConnectionStringBuilder.cs#L246). Judging by this [line in the SDK connection builder code](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/IotHubConnectionStringBuilder.cs#L367), you might need to establish a gateway hostname with an FQDN (i.e. [hostname].[domain]) for your Hostname parameter, instead of just the [hostname]. Let me know if that works.

**Comment by wvangeem** [comment]
@massand 
`openssl s_client -connect IOTDEMO07:8883 -CAfile C:/certificates/certs/azure-iot-test-only.root.ca.cert.pem -showcerts`
returns "Verify return code: 0 (ok)"
I already checked this before, and I think configuration and certificates are ok
`openssl s_client -connect IOTDEMO07.local:8883 -CAfile C:/certificates/certs/azure-iot-test-only.root.ca.cert.pem -showcerts`
returns also "Verify return code: 0 (ok)"
I can ping to IOTDEMO07 and IOTDEMO07.local

When I use connection string
`HostName=myhubname.azure-devices.net;DeviceId=iotdemo07-child01;SharedAccessKey=<MyKey>;GatewayHostName=iotdemo07`
this works, so I suspect configuration is ok.

So I changed now the connection string with FQDN, and indeed the check passes now
`HostName=iotdemo07.local;DeviceId=iotdemo07-child01;SharedAccessKey=<MyKey>`
The code now continues, but now I got following error
`AuthenticationException: TLS authentication error. The remote certificate is invalid according to the validation procedure.`
Same error now with connection string
`HostName=myhubname.azure-devices.net;DeviceId=iotdemo07-child01;SharedAccessKey=<MyKey>;GatewayHostName=iotdemo07.local`
Of course, my hostname in config.yaml is not iotdemo07.local, and this is one of the prerequisites.
I changed following setting in config.yaml
hostname: "iotdemo07.local"
and restart iotedge service.

And now the code works both with:
`HostName=myhubname.azure-devices.net;DeviceId=iotdemo07-child01;SharedAccessKey=<MyKey>;GatewayHostName=iotdemo07.local`
`HostName=iotdemo07.local;DeviceId=iotdemo07-child01;SharedAccessKey=<MyKey>`

So everything is ok now, thanks for your help.
Maybe we should adapt documentation that FQDN hostname should be used in connection string and config.yaml for downstream devices.
Best regards, Wim.
